6_survival_analysis/nbconverted/survival_hit_groups.py

```diff
@@ -0,0 +1,704 @@
+from sksurv.util import Surv
+
+import sys; sys.path.append('..')
+import config as cfg
+
+
+# In[2]:
+
+
+MPMP_LOCATION = Path('/home/jake/research/mpmp')
+
+# park et al. geneset info
+park_loss_data = cfg.data_dir / 'park_loss_df.tsv'
+park_gain_data = cfg.data_dir / 'park_gain_df.tsv'
+
+# park et al. significant gene info
+park_loss_sig_data = cfg.data_dir / 'park_loss_df_sig_only.tsv'
+park_gain_sig_data = cfg.data_dir / 'park_gain_df_sig_only.tsv'
+
+MIN_N_MUTATED = 0
+MAX_MIN_PVAL = 0.05
+
+
+# ### Load information for Park et al. genes
+
+# In[3]:
+
+
+park_loss_df = pd.read_csv(park_loss_data, sep='\t', index_col=0)
+park_loss_df.head()
+
+
+# In[4]:
+
+
+park_gain_df = pd.read_csv(park_gain_data, sep='\t', index_col=0)
+park_gain_df.head()
+
+
+# In[5]:
+
+
+park_loss_sig_df = pd.read_csv(park_loss_sig_data, sep='\t', index_col=0)
+park_loss_sig_df.head()
+
+
+# In[6]:
+
+
+park_gain_sig_df = pd.read_csv(park_gain_sig_data, sep='\t', index_col=0)
+park_gain_sig_df.head()
+
+
+# ### Get clinical endpoint info
+# 
+# Code for preprocessing clinical info is based on: https://github.com/greenelab/mpmp/blob/master/mpmp/utilities/data_utilities.py#L510
+# 
+# TODO: move this to a function/utilities file
+
+# In[7]:
+
+
+# use TCGA clinical data downloaded in mpmp repo
+clinical_filename = (
+    MPMP_LOCATION / 'data' / 'raw' / 'TCGA-CDR-SupplementalTableS1.xlsx'
+)
+
+
+# In[8]:
+
+
+clinical_df = pd.read_excel(
+    clinical_filename,
+    sheet_name='TCGA-CDR',
+    index_col='bcr_patient_barcode',
+    engine='openpyxl'
+)
+
+clinical_df.index.rename('patient_id', inplace=True)
+
+# drop numeric index column
+clinical_df.drop(labels=['Unnamed: 0'], axis=1, inplace=True)
+
+# we want to use age as a covariate
+clinical_df.rename(columns={'age_at_initial_pathologic_diagnosis': 'age'},
+                   inplace=True)
+
+print(clinical_df.shape)
+clinical_df.iloc[:5, :5]
+
+
+# In[9]:
+
+
+# we want to use overall survival as the target variable except for
+# certain cancer types where progression-free intervals are typically
+# used (since very few deaths are observed)
+# this is recommended in https://doi.org/10.1016/j.cell.2018.02.052
+pfi_cancer_types = [
+    'BRCA', 'DLBC', 'LGG', 'PCPG', 'PRAD',
+    'READ', 'TGCT', 'THCA', 'THYM'
+]
+
+clinical_df['time_in_days'] = clinical_df['OS.time']
+clinical_df['status'] = clinical_df['OS'].astype('bool')
+
+pfi_samples = clinical_df.type.isin(pfi_cancer_types)
+clinical_df.loc[pfi_samples, 'time_in_days'] = clinical_df[pfi_samples]['PFI.time']
+clinical_df.loc[pfi_samples, 'status'] = clinical_df[pfi_samples]['PFI'].astype('bool')
+
+# clean up columns and drop samples with NA survival times
+na_survival_times = (clinical_df['time_in_days'].isna())
+cols_to_keep = ['status', 'time_in_days', 'age', 'type']
+clinical_df = clinical_df.loc[~na_survival_times, cols_to_keep].copy()
+
+# mean impute missing age values
+clinical_df.age.fillna(clinical_df.age.mean(), inplace=True)
```

Comment:
You can impute ages? Wow!

Reply:
Probably poorly, though. I'm actually not using patient age in this script at all, so I'm going to remove this line.